src/compiler/diagnosticMessages.json

```diff
@@ -1939,6 +1939,10 @@
         "category": "Error",
         "code": 2689
     },
+    "'{0}' is not allowed to be used as a name of a static property or method in a class.": {

```

Comment:
this should be split into two error messages, one for properties and methods for target ES5, and one for properties only for ES6 and higher.